src/Service/PlanningService.php

```diff
@@ -0,0 +1,328 @@
+
+            if ($weekIsSupport) {
+                $supportWeek = new Weeks();
+                $supportWeek->weekCollection->add($week);
+                $supportWeek->weeks = 1;
+                $supportWeek->weekGoalLow = $this->weekGoalLow;
+                $supportWeek->weekGoalHigh = $this->weekGoalHigh;
+                $supportWeek->displayName = (string) $week;
+                $supportWeek->dateSpan = $firstAndLastDateOfWeek['first'].' - '.$firstAndLastDateOfWeek['last'];
+                if ($week == $currentWeek) {
+                    $supportWeek->activeSprint = true;
+                }
+                $weeks->add($supportWeek);
+            } else {
+                if (isset($regularWeek)) {
+                    $regularWeek->weekCollection->add($week);
+                    ++$regularWeek->weeks;
+                    $regularWeek->displayName .= '-'.$week;
+                    if ($week == $currentWeek) {
+                        $regularWeek->activeSprint = true;
+                    }
+
+                    if (3 === count($regularWeek->weekCollection)) {
+                        $regularWeek->dateSpan .= ' - '.$firstAndLastDateOfWeek['last'];
+                        $weeks->add($regularWeek);
+                        unset($regularWeek);
+                    }
+                } else {
+                    $regularWeek = new Weeks();
+                    $regularWeek->weekCollection->add($week);
+                    $regularWeek->weeks = 1;
+                    $regularWeek->weekGoalLow = $this->weekGoalLow * 3;
+                    $regularWeek->weekGoalHigh = $this->weekGoalHigh * 3;
+                    $regularWeek->displayName = (string) $week;
+                    $regularWeek->dateSpan = $firstAndLastDateOfWeek['first'];
+                    if ($week == $currentWeek) {
+                        $regularWeek->activeSprint = true;
+                    }
+                }
+            }
+        }
+
+        return $weeks;
+    }
+
+    /**
+     * Sorts issues by week.
+     *
+     * @param array $allIssues the array of all issues to sort
+     *
+     * @return array the sorted array of issues
+     */
+    private function sortIssuesByWeek(array $allIssues): array
+    {
+        $weekIssues = [];
+
+        foreach ($allIssues as $issue) {
+            $issueDueDate = $issue->getDueDate();
+
+            if (!$issueDueDate) {
+                continue;
+            }
+
+            $issueYear = $issueDueDate->format('Y');
+            $issueWeek = (int) $issueDueDate->format('W');
+
+            if ($issueYear) {
+                $weekIssues[$issueWeek][] = $issue;
+            } else {
+                $weekIssues['unscheduled'][] = $issue;
+            }
+        }
+
+        return $weekIssues;
+    }
+
+    private function processIssuesForWeek(PlanningData $planning, int $week, array $issues): void
+    {
+        foreach ($issues as $issueData) {
+            if ('0' !== $issueData->getStatus()) {
```

Comment:
What statuses are possible here? '0' is a weird value. The best would be that getStatus() returns an enum

Reply:
I will implement this on a seperate branch to break it up a bit.